security/pkg/pki/ca/ca.go

```diff
@@ -136,6 +137,38 @@ func NewSelfSignedIstioCAOptions(caCertTTL, certTTL, maxCertTTL time.Duration, o
 	return opts, nil
 }
 
+// NewIstioCAOptions returns a new IstioCAOptions instance using given certificate.
+func NewIstioCAOptions(certChainFile, signingCertFile, signingKeyFile, rootCertFile string,
+	certTTL, maxCertTTL time.Duration) (*IstioCAOptions, error) {
+	caOpts := &IstioCAOptions{
+		CertTTL:    certTTL,
+		MaxCertTTL: maxCertTTL,
+	}
+	if certChainFile != "" {
+		if certChainBytes, err := readFile(certChainFile); err != nil {
```

Comment:
should the logic be opposite? 

if err != nil, we should "return nil, err" instead, right?

Reply:
Ah, you are right. Thanks.

Reply:
I'll add a test for this.